$NetBSD: patch-configure.in,v 1.1.2.1 2012/10/10 13:48:13 tron Exp $

* Use separate @LIBREADLINE@ AC_SUBST.
* Avoid using "==" for argument of test(1).

--- configure.in.orig	2012-09-27 00:35:19.000000000 +0000
+++ configure.in
@@ -157,7 +157,7 @@ case "$use_python" in
 			AC_PATH_PROGS(PYTHON, $use_python)
 			;;
 		esac
-		if test "X$PYTHON" == "X"
+		if test "X$PYTHON" = "X"
 		then
 			case "$use_python" in
 			unspec)
@@ -2322,23 +2322,18 @@ AC_SUBST(ISC_PLATFORM_NEEDSTRLCAT)
 
 AC_ARG_WITH(readline,
         [  --with-readline[=LIBSPEC]    specify readline library [default -lreadline]],
-        readline="$withval", readline="-lreadline")
+        readline="$withval", readline="readline")
 case "$readline" in
 no)	;;
 *)
 	if test "x$readline" = "xyes"
 	then
-		readline=-lreadline
-	fi
-	saved_LIBS="$LIBS"
-	LIBS="$LIBS $readline"
-	AC_CHECK_FUNCS(readline)
-	if test "$ac_cv_func_readline" = "no" 
-	then
-		LIBS="$saved_LIBS"
+		readline=readline
 	fi
+	AC_CHECK_LIB($readline, [readline], LIBREADLINE=-lreadline)
         ;;
 esac
+AC_SUBST(LIBREADLINE)
 
 
 ISC_PRINT_OBJS=
